bokeh/core/property/factory.py

```diff
@@ -0,0 +1,108 @@
+''' Provide a Base class for all Bokeh properties.
+
+Bokeh properties work by contributing Python descriptor objects to
+``HasProps`` classes. These descriptors then delegate attribute access back
+to the Bokeh property class, which handles validation, serialization, and
+documentation needs.
+
+The ``PropertyFactory`` class provides two methods ``autocreate`` and
+``make_descriptors`` that are used by the metaclass ``MetaHasProps`` during
+class creation. to install the necessary descriptors corresponding to the
+declared properties.
+
+This machinery helps to make Bokeh much more user friendly. For example,
+the DataSpec properties mediate between fixed values and references to column
+data source columns. A user can use a very simple syntax, and the property
+will correctly serialize and validate automatically:
+
+.. code-block:: python
+
+    from bokeh.models import Circle
+
+    c = Circle()
+
+    c.x = 10      # serializes to {'value': 10}
+
+    c.x = 'foo'   # serializes to {'field': 'foo'}
+
+    c.x = [1,2,3] # raises a ValueError validation exception
+
+There are many other examples like this throughout Bokeh. In this way users
+may operate simply and naturally, and not be concerned with the low-level
+details around validation, serialization, and documentation.
+
+.. note::
+    These classes form part of the very low-level machinery that implements
+    the Bokeh model and property system. It is unlikely that any of these
+    classes or their methods will be applicable to any standard usage or to
+    anyone who is not directly developing on Bokeh's own infrastructure.
+
+'''
+
+class PropertyFactory(object):
```

Comment:
This used to have a subtype PropertyDescriptor which manufactured properties, and now it has a subtype Property which manufactures PropertyDescriptor. If doing that, surely this class should be renamed PropertyDescriptorFactory?